drivers/wifi/esp_at/esp_offload.c

```diff
@@ -405,75 +426,53 @@ static int esp_send(struct net_pkt *pkt,
 	return esp_sendto(pkt, NULL, 0, cb, timeout, user_data);
 }
 
-#define CIPRECVDATA_CMD_MIN_LEN (sizeof("+CIPRECVDATA,L:") - 1)
-#define CIPRECVDATA_CMD_MAX_LEN (sizeof("+CIPRECVDATA,LLLL:") - 1)
-
-static int cmd_ciprecvdata_parse(struct esp_socket *sock,
-				 struct net_buf *buf, uint16_t len,
-				 int *data_offset, int *data_len)
+MODEM_CMD_DEFINE(on_cmd_ciprecvdata)
 {
-	char cmd_buf[CIPRECVDATA_CMD_MAX_LEN + 1];
-	char *endptr;
-	size_t frags_len;
-	size_t match_len;
-
-	frags_len = net_buf_frags_len(buf);
-	if (frags_len < CIPRECVDATA_CMD_MIN_LEN) {
-		return -EAGAIN;
-	}
+	struct esp_data *dev = CONTAINER_OF(data, struct esp_data,
+					    cmd_handler_data);
+	struct esp_socket *sock = dev->rx_sock;
+	int err;
 
-	match_len = net_buf_linearize(cmd_buf, CIPRECVDATA_CMD_MAX_LEN,
-				      buf, 0, CIPRECVDATA_CMD_MAX_LEN);
-	cmd_buf[match_len] = 0;
+	int socket_data_length = atoi(argv[0]);
 
-	*data_len = strtol(&cmd_buf[len], &endptr, 10);
-	if (endptr == &cmd_buf[len] ||
-	    (*endptr == 0 && match_len >= CIPRECVDATA_CMD_MAX_LEN) ||
-	    *data_len > CIPRECVDATA_MAX_LEN) {
-		LOG_ERR("Invalid cmd: %s", cmd_buf);
-		return -EBADMSG;
-	} else if (*endptr == 0) {
+	if ((net_buf_frags_len(data->rx_buf) - 2) < socket_data_length) {
+		LOG_DBG("Not enough data -- wait!");
 		return -EAGAIN;
-	} else if (*endptr != _CIPRECVDATA_END) {
-		LOG_ERR("Invalid end of cmd: 0x%02x != 0x%02x", *endptr,
-			_CIPRECVDATA_END);
-		return -EBADMSG;
 	}
 
-	/* data_offset is the offset to where the actual data starts */
-	*data_offset = (endptr - cmd_buf) + 1;
-
-	/* FIXME: Inefficient way of waiting for data */
-	if (*data_offset + *data_len > frags_len) {
-		return -EAGAIN;
-	}
+	struct sockaddr_in *recv_addr =
+			(struct sockaddr_in *) &sock->context->remote;
+	int port = atoi(argv[2]);
 
-	*endptr = 0;
+	recv_addr->sin_port = ntohs(port);
+	recv_addr->sin_family = AF_INET;
 
-	return 0;
-}
+	/* IP addr comes within quotation marks, which is disliked by
+	 * conv function. So we remove them by subtraction 2 from argv[1]
+	 * length and index from &argv[1][1].
+	 */
+	char remote_ip_addr[INET_ADDRSTRLEN];
+	size_t remote_ip_str_len;
 
-MODEM_CMD_DIRECT_DEFINE(on_cmd_ciprecvdata)
-{
-	struct esp_data *dev = CONTAINER_OF(data, struct esp_data,
-					    cmd_handler_data);
-	struct esp_socket *sock = dev->rx_sock;
-	int data_offset, data_len;
-	int err;
+	remote_ip_str_len = MIN(sizeof(remote_ip_addr) - 1, strlen(argv[1]) - 2);
+	if (remote_ip_str_len > sizeof(remote_ip_addr) - 1) {
+		LOG_ERR("Received IP-address string it too long.");
+		err = -EIO;
+		return err;
+	}
```

Comment:
This check is no-op, the `remote_ip_str_len` cannot be larger that the sizeof() as we just set the minimum value in previous line. You can remove the check, it is not really needed here.

Reply:
Removed it.